oauth2_http/javatests/com/google/auth/oauth2/ServiceAccountCredentialsTest.java

```diff
@@ -292,6 +293,34 @@ public void refreshAccessToken_refreshesToken() throws IOException {
   }
 
   @Test
+  public void refreshAccessToken_tokenExpiry() throws IOException {
+    final String tokenString = "1/MkSJoj1xsli0AccessToken_NKPY2";
+    MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
+    MockTokenServerTransport transport = transportFactory.transport;
+    ServiceAccountCredentials credentials =
+        ServiceAccountCredentials.fromPkcs8(
+            SA_CLIENT_ID,
+            SA_CLIENT_EMAIL,
+            SA_PRIVATE_KEY_PKCS8,
+            SA_PRIVATE_KEY_ID,
+            SCOPES,
+            transportFactory,
+            null);
+    credentials.clock = new FixedClock(0L);
+
+    transport.addServiceAccount(SA_CLIENT_EMAIL, tokenString);
+    AccessToken accessToken = credentials.refreshAccessToken();
+    assertEquals(tokenString, accessToken.getTokenValue());
+    assertEquals(3600 * 1000L, accessToken.getExpirationTimeMillis().longValue());
+
+    // Test for large expires_in values (should not overflow).
+    transport.setExpiresInSeconds(3600 * 1000);
```

Comment:
`setExpiresInSeconds()` takes an int. In `MockTokenServerTransport`, this was set to the constant int value 3600. I kept it as an int when adding the `setExpiresInSeconds()` method.